core/src/date.ts

```diff
@@ -4,13 +4,18 @@ import { getNetworkConfigByType } from './constants/NETWORKS';
 
 const CURRENT_BLOCK_DATE_CACHE_EXPIRY_MS = 60 * 1000;
 
-const fetchLatestBlockDate = async function (network: string): Promise<Date> {
+export const fetchDateByBlockNumber = async function (network: string, blockNumber: number): Promise<Date> {
     const provider = await getProvider(network);
-    const blockNumber = await provider.getBlockNumber();
     const block = await provider.getBlock(blockNumber);
     return new Date(block.timestamp * 1000);
 };
 
+const fetchLatestBlockDate = async function (network: string): Promise<Date> {
```

Comment:
Do we still want this function, or should I remove it?

Reply:
`fetchLatestBlockDate` is used in other places, you just refactored it into two functions, which is ok to keep.

Also, [the UI proposal](https://github.com/sidestream-tech/unified-auctions-ui/pull/90#issuecomment-1048898985) actually have `at ${takeEvents[takeEvents.length - 1].date.toISOString()}` which you may want to add to the text as a bonus (but then you would need to fetch date for all events and change `Event` type to our custom `AuctionEvent` which contains the date)

Reply:
Added this dates!

Reply:
I personally prefer displaying dates with `toDateString()` as I believe it is more readable. 
Is there a reason you wanted to use `.toISOString()` or can I use `toDateString()`?